packages/babel-plugin-proposal-decorators/src/transformer-2021-12.ts

```diff
@@ -879,6 +875,7 @@ function transformClass(
   const originalClass = path.node;
 
   if (classDecorators) {
+    locals.push(classLocal, classInitLocal);
```

Comment:
`classLocal` and `classInitLocal` are generated after other functions, see `applyClassDecs` in `applyDecs`.